subsys/net/lib/lwm2m/lwm2m_engine.c

```diff
@@ -5707,6 +5839,18 @@ int lwm2m_socket_start(struct lwm2m_ctx *client_ctx)
 			goto error;
 		}
 
+		session_cache = IS_ENABLED(CONFIG_LWM2M_TLS_SESSION_CACHING) ?
+					   TLS_SESSION_CACHE_ENABLED :
+					   TLS_SESSION_CACHE_DISABLED;
+
+		ret = setsockopt(client_ctx->sock_fd, SOL_TLS, TLS_SESSION_CACHE, &session_cache,
+				 sizeof(session_cache));
+		if (ret < 0) {
+			ret = -errno;
+			LOG_ERR("Failed to set TLS_SESSION_CACHE option: %d", errno);
+			goto error;
+		}
```

Comment:
```suggestion
		if (IS_ENABLED(CONFIG_LWM2M_TLS_SESSION_CACHING)) {
			int session_cache = TLS_SESSION_CACHE_ENABLED;

						ret = setsockopt(client_ctx->sock_fd, SOL_TLS, TLS_SESSION_CACHE, &session_cache,
					 sizeof(session_cache));
			if (ret < 0) {
				ret = -errno;
				LOG_ERR("Failed to set TLS_SESSION_CACHE option: %d", errno);
				goto error;
			}
		}
```

I would prefer you not to touch the value, unless enabled.

Reply:
OK I'm fine for this change.